lib/rawlink.ml

```diff
@@ -49,7 +49,7 @@ let close_link t = Unix.close t.fd
 
 let send_packet t buf =
   let len = Cstruct.len buf in
-  let n = Unix.write t.fd (Cstruct.to_string buf) 0 len in
+  let n = Unix.write t.fd (Cstruct.to_string buf |> Bytes.of_string) 0 len in
```

Comment:
btw, another option is to depend on `cstruct 3.2.0` and use the new `Cstruct.to_bytes`!